src/_P184_Triac.ino

```diff
@@ -0,0 +1,380 @@
+        timerAttachInterrupt(P184_data.p184_timer, &P184_timer_handler);
+        timerAlarm(P184_data.p184_timer, P184_data.time_us, true, 0);
+      }
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      // code to be executed to read data
+      UserVar.setFloat(event->TaskIndex, 0, P184_data.trigger_value);
+      UserVar.setFloat(event->TaskIndex, 1, P184_data.power_value);
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WRITE:
+    {
+
+      // parse string to extract the command
+      String tmpString = parseString(string, 1); // already converted to lowercase
+
+      if (equals(tmpString, F("triac"))) {
+        String subcmd = parseString(string, 2);
+        String valueStr = parseString(string, 3);
+        long value = valueStr.toInt();
+
+        if (value >= 0 && value <= 100) {
+          if (equals(subcmd, F("power"))) {
+            // User wants to set POWER: triac,power,<value>
+            P184_data.power_value = value;
+            // Find the corresponding trigger value from LUT
+            uint8_t new_trigger = pgm_read_byte(&power_to_trigger_lut[P184_data.power_value]);
+            P184_data.trigger_value = new_trigger;
+            success = true;
+          } else if (equals(subcmd, F("trigger"))) {
+            // User wants to set TRIGGER directly: triac,trigger,<value>
+            uint8_t new_trigger = value;
+
+            // Apply dead zone logic
+            if (new_trigger > (100 - P184_data.dead_zone)) {
+              new_trigger = 100;
+            } else if (new_trigger < P184_data.dead_zone) {
+              new_trigger = 0;
+            }
+            P184_data.trigger_value = new_trigger;
+
+            // Let's find the closest power value for the new trigger.
+            // This is a slow lookup, but only happens on command.
+            // The LUT maps power (index) to trigger (value). We need to find the index (power)
+            // for a given trigger value.
+            for (int i = 0; i <= 100; ++i) {
+              // Find the first power level (i) where the corresponding trigger
+              // is less than or equal to the one we just set.
+              if (pgm_read_byte(&power_to_trigger_lut[i]) <= P184_data.trigger_value) {
+                P184_data.power_value = i;
+                break; // Found the highest power for this trigger level or lower
+              }
+            }
+            success = true;
+          }
+
+          if (success) {
+            P184_TRIGGER_CONFIG() = P184_data.trigger_value; // Save state
+            P184_data.time_us = uint64_t(P184_data.trigger_value) * P184_data.freq_timing_val;
+            timerAlarm(P184_data.p184_timer, P184_data.time_us, true, 0);
+            if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+              String log = strformat(F("P184 CMD : Trigger %d%% . Power %d%%"), P184_data.trigger_value, P184_data.power_value);
+              addLogMove(LOG_LEVEL_INFO, log);
```

Comment:
This can be a single statement, no need to instantiate a new String, only to pass it on.
Then you can also use `addLog` instead of `addLogMove`.